hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -404,6 +396,52 @@ public void refreshTimeline() throws IOException {
     return sourceDataToSync;
   }
 
+  public boolean isDataAvailableForIngestion() throws IOException {
```

Comment:
Alternate approach for doing this can be exposing another abstract method `isDataAvailable()` in `Source` class with each source implementing its own logic. 

The current way seems the easier way of doing this without much code changes. Please let me know if you approve of this @nsivabalan @vinothchandar ?